src/sys/fuchsia/eventedfd.rs

```diff
@@ -0,0 +1,187 @@
+use {io, poll, Evented, Ready, Poll, PollOpt, Token};
+use libc;
+use magenta;
+use magenta::HandleBase;
+use sys::fuchsia::{DontDrop, poll_opts_to_wait_async, sys};
+use std::mem;
+use std::os::unix::io::RawFd;
+use std::sync::{Arc, Mutex};
+
+/// Properties of an `EventedFd`'s current registration
+#[derive(Debug)]
+pub(in sys::fuchsia) struct EventedFdRegistration {
+    pub token: Token,
+    pub handle: DontDrop<magenta::Handle>,
+    pub rereg_signals: Option<(magenta::Signals, magenta::WaitAsyncOpts)>,
+}
+
+/// An event-ed file descriptor. The file descriptor is owned by this structure.
+#[derive(Debug)]
+pub(in sys::fuchsia) struct EventedFdInner {
+    /// Properties of the current registration.
+    pub registration: Mutex<Option<EventedFdRegistration>>,
+
+    /// Owned file descriptor.
+    pub fd: RawFd,
```

Comment:
It _could_, but since the `Selector` keeps a map of references to `EventedFd`s for different types (`TcpSocket`, `TcpStream`, `UdpSocket`), we'd need to figure out how to handle that case. Perhaps an enum of `TcpSocket`, `TcpStream`, `UdpSocket` would work. What do you think?